lib/elixir/lib/exception.ex

```diff
@@ -454,44 +454,52 @@ defmodule Exception do
   end
 
   defp format_sup_data({:invalid_intensity, intensity}) do
-    "invalid intensity: " <> inspect(intensity)
+    "invalid max_restarts (intensity): " <> inspect(intensity)
   end
 
   defp format_sup_data({:invalid_period, period}) do
-    "invalid period: " <> inspect(period)
+    "invalid max_seconds (period): " <> inspect(period)
   end
 
-  defp format_sup_data(other), do: inspect(other)
+  defp format_sup_data(other), do: "got: #{inspect other}"
+
+  defp format_sup_spec({:duplicate_child_name, id}) do
+   """
+   more than one child specification has the id: #{inspect id}
+   If using maps as child specifications, make sure the IDs are unique.
+   If using a module or {module, arg}, use Supervisor.child_spec/3.
+   """
```

Comment:
The error message is saying exactly how to solve it? Use Supervisor.child_spec/2:

    children = [
      Supervisor.child_spec(Registry, id: 1),
      Supervisor.child_spec(Registry, id: 2)
    ]

?